src/ui/MainWindow.cc

```diff
@@ -368,30 +370,22 @@ void MainWindow::_showDockWidget(const QString& name, bool show)
 void MainWindow::_createInnerDockWidget(const QString& widgetName)
 {
     QGCDockWidget* widget = NULL;
+    QAction *action = _mapName2Action[widgetName];
     
-    if (widgetName == _mavlinkDockWidgetName) {
-        widget = new QGCMAVLinkInspector(widgetName, _mapName2Action[widgetName], MAVLinkProtocol::instance(),this);
-    } else if (widgetName == _customCommandWidgetName) {
-        widget = new CustomCommandWidget(widgetName, _mapName2Action[widgetName], this);
-    } else if (widgetName == _filesDockWidgetName) {
-        widget = new QGCUASFileViewMulti(widgetName, _mapName2Action[widgetName], this);
-    } else if (widgetName == _uasStatusDetailsDockWidgetName) {
-        widget = new UASInfoWidget(widgetName, _mapName2Action[widgetName], this);
-    } else if (widgetName == _pfdDockWidgetName) {
-        widget = new FlightDisplayWidget(widgetName, _mapName2Action[widgetName], this);
-    } else if (widgetName == _hilDockWidgetName) {
-        widget = new HILDockWidget(widgetName, _mapName2Action[widgetName], this);
-    } else if (widgetName == _analyzeDockWidgetName) {
-        widget = new Linecharts(widgetName, _mapName2Action[widgetName], mavlinkDecoder, this);
-    } else if (widgetName == _uasInfoViewDockWidgetName) {
-        QGCTabbedInfoView* pInfoView = new QGCTabbedInfoView(widgetName, _mapName2Action[widgetName], this);
-        pInfoView->addSource(mavlinkDecoder);
-        widget = pInfoView;
-    } else {
-        qWarning() << "Attempt to create unknown Inner Dock Widget" << widgetName;
-	return;
+    switch(action->data().toInt()) {
+    case MAVLINK_INSPECTOR: widget = new QGCMAVLinkInspector(widgetName, action, MAVLinkProtocol::instance(),this); break;

```

Comment:
okay. commit is ready, if you wanna merge this, the other will be done in a minute.